release_dashboard/lib/widgets/create_release_substeps.dart

```diff
@@ -112,30 +140,67 @@ class CreateReleaseSubstepsState extends State<CreateReleaseSubsteps> {
 }
 
 typedef SetReleaseData = void Function(String name, String data);
+typedef ChangeIsEachInputValid = void Function(int index, bool isValid);
 
 /// Captures the input values and updates the corresponding field in [releaseData].
 class InputAsSubstep extends StatelessWidget {
-  const InputAsSubstep({
+  InputAsSubstep({
     Key? key,
     required this.index,
     required this.setReleaseData,
     this.hintText,
+    required this.changeIsInputValid,
   }) : super(key: key);
 
   final int index;
   final SetReleaseData setReleaseData;
   final String? hintText;
+  final ChangeIsEachInputValid changeIsInputValid;
 
   @override
   Widget build(BuildContext context) {
+    late RegExp formRegexValidator;
+    // assigns corresponding regex to each input
+    switch (index) {
```

Comment:
I think rather than hard-coding into this widget the behavior and order of the different fields, it would probably be better to create different classes for the different types of input.

Reply:
On a higher level, as part of encapsulation, a widget probably should not need to know its position relative to its siblings.

Reply:
Yes, create a different widget for each different "type" of input.